### PR TITLE
Add reducer template

### PR DESCRIPTION
**Pull request by bitomule**
This PR adds a reducer file template and also adds it to the scene template.

### CODE REVIEW

**Review by JosepEscobar — APPROVED** [review]

File: Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import ComposableArchitecture
+
+/// sourcery: AutoMockable
+protocol ___FILEBASENAME___Delegate: AnyObject {
+    func present(_ state: ___FILEBASENAME___.State)
+}
+
+final class ___FILEBASENAME___: Reducer {
+    struct State {
+
```
Comment (Machipla) [inline Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift:10]: Remove for not complaining on swiftlint on first instance?
```suggestion
```

File: Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    func present(_ state: ___FILEBASENAME___.State)
+}
+
+final class ___FILEBASENAME___: Reducer {
+    struct State {
+        
+    }
+
+    enum Actions: Action {
+
```
Comment (Machipla) [inline Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift:14]: same
```suggestion
```

File: Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    func reduce(_ state: inout ___FILEBASENAME___.State, _ action: Actions) -> Effect<Actions> {
+        .none
+    }
+}
+
+extension ___FILEBASENAME___ {
+    struct Effects {
+        init() {
+            // Inject usecases
```
Comment (Machipla) [inline Wallapop Architecture/Reducer.xctemplate/___FILEBASENAME___.swift:32]: What about doing this as a completion text?
```suggestion
            <#Inject usecases#>
```

File: Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import ComposableArchitecture
+
+/// sourcery: AutoMockable
+protocol ___FILEBASENAME___Delegate: AnyObject {
+    func present(_ state: ___FILEBASENAME___.State)
+}
+
+final class ___FILEBASENAME___: Reducer {
+    struct State {
+
```
Comment (Machipla) [inline Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift:10]:
```suggestion
```

File: Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    func present(_ state: ___FILEBASENAME___.State)
+}
+
+final class ___FILEBASENAME___: Reducer {
+    struct State {
+        
+    }
+
+    enum Actions: Action {
+
```
Comment (Machipla) [inline Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift:14]:
```suggestion
```

File: Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    func reduce(_ state: inout ___FILEBASENAME___.State, _ action: Actions) -> Effect<Actions> {
+        .none
+    }
+}
+
+extension ___FILEBASENAME___ {
+    struct Effects {
+        init() {
+            // Inject usecases
```
Comment (Machipla) [inline Wallapop Architecture/Scene.xctemplate/___FILEBASENAME___Reducer.swift:32]:
```suggestion
            <#Inject usecases#>
```

**Review by Machipla — APPROVED** [review]
LGTM!

**Review by Polenoso — APPROVED** [review]

**Comment by bitomule** [comment]
Thanks, @Machipla, applied your commit suggestions!

**Review by jositocuadrado — APPROVED** [review]

**Review by mdellagi77 — APPROVED** [review]
Uff que guay

**Review by PabloDomine — APPROVED** [review]

**Review by carguezu — APPROVED** [review]
Genial!!! 👏👏👏

**Review by davidmmartinez — APPROVED** [review]

**Review by HerdzikWallapop — APPROVED** [review]

**Review by filip4Wall — APPROVED** [review]

**Review by josepbordesjove — APPROVED** [review]

**Review by MoralAlberto — APPROVED** [review]

**Review by manuel-martinez84 — APPROVED** [review]

**Review by carlesgrauwallapop — APPROVED** [review]

**Pull request merged**